### DISCUSSION TITLE
Weird text selecting issue

### DISCUSSION THREAD

**Discussion by glaprida** [GrapesJS/grapesjs#5517] [Q&A] [answered]
I noticed that once a text component is activated (double click, so that the RTE is shown) if you select a portion of text (dragging the cursor) but you release it outside the text component it does some weird behaviour: you can still edit the text, but the selected component is the one you released the mouse on.

It happens on the current GrapesJs demo.

I would like a way either:

A) prevent the component where the mouse is released to be selected (preferred option)

or B) to 'deactivate' the text component so that you can not edit once the other component is selected.

**Comment by glaprida** [comment, marked as answer]
This was my solution:

```
editor.on('component:selected', (c)=>{
 
        if(editor.getSelected()!=undefined && editor.getEditing()!=undefined){
            if(editor.getSelected().cid != editor.getEditing().cid){
                setTimeout(()=>{ // you might not need the timeout. in my case it was required for other plugin to work right
                    editor.select(editor.getEditing())
                },1)
            }
        }
});
```

It basically re-select the element that's being edited.

**Reply by artf** [reply]
Makes sense to prevent this at the core level, I'll add it.
